2023/adv2023-rust/src/day11.rs

```diff
@@ -14,65 +22,128 @@ fn main() {
 
     // Find coords of galaxies
     println!("Find coords of galaxies");
-    let mut galaxies = Vec::new();
+    let galaxies = find_galaxies(&img);
 
-    for i in 0..img.len() {
-        for j in 0..img[0].len() {
-            if img[i][j] == '#' {
-                galaxies.push((i, j));
-            }
-        }
-    }
+    // Find distances between each pair of galaxies
+    println!("Find distances between each pair of galaxies");
+    // let mut total_dist = 0;
+    // for x in 0..galaxies.len() {
+    //     for y in x+1..galaxies.len() {
+    //         let src = galaxies[x];
+    //         let dest = galaxies[y];
+
+    //         // Manhattan distance
+    //         total_dist += src.0.abs_diff(dest.0) + src.1.abs_diff(dest.1);
+    //     }
+    // }
+    // println!("  {:?}", distances);
+    // println!("  {:?}", distances.values());
+
+    let total_dist = galaxies
+        .iter()
+        .enumerate()
+        .flat_map(|(i, &src)| galaxies
+            .iter()
+            .skip(i + 1)
+            .map(move |&dest| src.0.abs_diff(dest.0) + src.1.abs_diff(dest.1))
+        )
+        .sum();
+
+    return total_dist;
+}
+
+fn part2() -> usize {
```

Comment:
This could take `expansion_factor` as an arg, and handle both part1 and part2 that way. :)